test/integration/build/test_build.py

```diff
@@ -80,6 +81,272 @@ def test_app_without_rockspec(cartridge_cmd, project_without_dependencies):
+    os.mkdir(dir_path)
+
+    version = 'scm-2'
+    rockspec_name = '{}-{}.rockspec'.format(project.name, version)
+    rockspec_path = os.path.join(dir_path, rockspec_name)
+
+    with open(rockspec_path, 'w') as f:
+        f.write('''
+                package = '{}'
+                version = '{}'
+                source  = {{ url = '/dev/null' }}
+                dependencies = {{ 'tarantool' }}
+                build = {{ type = 'none' }}
+            '''.format(project.name, version))
+
+    build_logs = [
+        'Build application in',
+        'Running `cartridge.pre-build`',
+        'Running `tarantoolctl rocks make {}`'.format(rockspec_path),
+        'Application was successfully built',
+    ]
+
+    # with --spec and .rockspec file from other directory
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        rockspec_path,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+
+
+def test_building_with_one_rockspec_in_project_root(cartridge_cmd, project_without_dependencies):
+    # with *-scm-1.rockspec in project root
+    project = project_without_dependencies
+    prebuild_output = "pre-build hook output"
+
+    with open(os.path.join(project.path, 'cartridge.pre-build'), 'w') as f:
+        prebuild_script_lines = [
+            "#!/bin/sh",
+            "echo \"{}\"".format(prebuild_output)
+        ]
+        f.write('\n'.join(prebuild_script_lines))
+
+    build_logs = [
+        'Build application in',
+        'Running `cartridge.pre-build`',
+        'Running `tarantoolctl rocks make`',
+        'Application was successfully built',
+    ]
+
+    # without --spec
+    cmd = [
+        cartridge_cmd,
+        "build",
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+    assert prebuild_output not in output
+
+    # check that all expected rocks was installed
+    files = recursive_listdir(project.path)
+    assert '.rocks' in files
+    assert all([rock in files for rock in project.rocks_content])
+
+    build_logs = [
+        'Build application in',
+        'Running `cartridge.pre-build`',
+        'Running `tarantoolctl rocks make {}`'.format(project.rockspec_name),
+        'Application was successfully built',
+    ]
+
+    # with --spec
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        project.rockspec_name,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+    assert prebuild_output not in output
+
+    # check that all expected rocks was installed
+    files = recursive_listdir(project.path)
+    assert '.rocks' in files
+    assert all([rock in files for rock in project.rocks_content])
+
+    # with *-scm-2.rockspec in project root
```

Comment:
This test isn't a test of `cartridge build`. In fact, you are testing `rocks make`. But you should test CLI and it's responsibility is to pass correct spec to `rocks make`.

I suggest you removing this test. Building application with one rockspec without `--spec` specifying is a default case that is already tested.